llm_on_ray/finetune/finetune.py

```diff
@@ -193,92 +242,43 @@ def train_func(config: Dict[str, Any]):
         }
     )
 
-    optimizer = common.optimizer.Optimizer.registory.get("DefaultOptimizer")()(
-        model,
-        config={
-            "name": config["Training"]["optimizer"],
-            "config": {"lr": config["Training"]["learning_rate"]},
-        },
+    data_collator = common.dataprocesser.general_processer.DataCollatorForCompletionOnlyLM(
+        tokenizer=tokenizer, mlm=False, return_tensors="pt", pad_to_multiple_of=8
     )
 
-    accelerate_mode = config["Training"]["accelerate_mode"]
-    if accelerate_mode == "FSDP":
-        fsdp_plugin = FullyShardedDataParallelPlugin(
-            state_dict_config=FullStateDictConfig(offload_to_cpu=False, rank0_only=False),
-            optim_state_dict_config=FullOptimStateDictConfig(
-                offload_to_cpu=False, rank0_only=False
-            ),
+    if device in ["cpu", "gpu"]:
+        from transformers import Trainer, TrainingArguments
+
+        training_args = convert_to_training_args(TrainingArguments, config)
+        trainer = Trainer(
+            model=model,
+            args=training_args,
+            train_dataset=tokenized_datasets["train"],
+            eval_dataset=tokenized_datasets["validation"],
+            tokenizer=tokenizer,
+            data_collator=data_collator,
```

Comment:
I think we should pass optimizer parameter into the Trainer according to the user's configuration.

Reply:
optimizer's parameters are parts of `TrainingArguments`, and optimizer will be create before calling epoch-step-loop in Trainer.train()

Reply:
But I don’t see the optimizer configured in TrainingArguments. How does user set optimizer parameter?